From khali@linux-fr.org Mon Aug 28 05:18:20 2006
Date: Mon, 28 Aug 2006 14:18:14 +0200
From: Jean Delvare <khali@linux-fr.org>
To: Greg KH <greg@kroah.com>
Cc: LM Sensors <lm-sensors@lm-sensors.org>, Alexey Dobriyan <adobriyan@gmail.com>
Subject: [PATCH 01/14] atxp1: Signed/unsigned char bug fix
Message-Id: <20060828141814.71640724.khali@linux-fr.org>
Content-Disposition: inline; filename=hwmon-atxp1-signed-unsigned-char-bug.patch

From: Alexey Dobriyan <adobriyan@gmail.com>

vid_to_reg() can return -1 and char can be unsigned.

Signed-off-by: Alexey Dobriyan <adobriyan@gmail.com>
Signed-off-by: Jean Delvare <khali@linux-fr.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/hwmon/atxp1.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- gregkh-2.6.orig/drivers/hwmon/atxp1.c
+++ gregkh-2.6/drivers/hwmon/atxp1.c
@@ -116,8 +116,7 @@ static ssize_t atxp1_storevcore(struct d
 {
 	struct atxp1_data *data;
 	struct i2c_client *client;
-	char vid;
-	char cvid;
+	int vid, cvid;
 	unsigned int vcore;
 
 	client = to_i2c_client(dev);
